luigi/task.py

```diff
@@ -324,7 +337,19 @@ def __hash__(self):
         return self.__hash
 
     def __repr__(self):
```

Comment:
I've tried this but the results are inconsistent in the tests.  I'm seeing a mixture of `param='1'` and `param=1` for `IntParameter`s.  I'm not sure what's going on but I think keeping `str(task)` the same for backward compatibility is the pragmatic choice.